core/src/test/java/org/keycloak/sdjwt/SdJwtFacadeTest.java

```diff
@@ -0,0 +1,126 @@
+package org.keycloak.sdjwt;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.keycloak.common.VerificationException;
+import org.keycloak.crypto.SignatureSignerContext;
+import org.keycloak.crypto.SignatureVerifierContext;
+import org.keycloak.rule.CryptoInitRule;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for SdJwtFacade.
+ *
+ * @author <a href="mailto:rodrick.awambeng@adorsys.com">Rodrick Awambeng</a>
+ */
+public abstract class SdJwtFacadeTest {
+
+    @ClassRule
+    public static CryptoInitRule cryptoInitRule = new CryptoInitRule();
+
+    private static final String HASH_ALGORITHM = "sha-256";
+    private static final String JWS_TYPE = "JWS_TYPE";
+
+    private SdJwtFacade sdJwtFacade;
+
+    private JsonNode claimSet;
+    private DisclosureSpec disclosureSpec;
+
+    @Before
+    public void setUp() {
+        SignatureSignerContext signer = TestSettings.getInstance().getIssuerSignerContext();
+
+        sdJwtFacade = new SdJwtFacade(signer, HASH_ALGORITHM, JWS_TYPE);
+
+        claimSet = TestUtils.readClaimSet(getClass(), "sdjwt/a1.example2-holder-claims.json");
+        disclosureSpec = DisclosureSpec.builder()
+                .withUndisclosedClaim("sub", "2GLC42sKQveCfGfryNRN9w")
+                .withUndisclosedClaim("given_name", "eluV5Og3gSNII8EYnsxA_A")
+                .withUndisclosedClaim("family_name", "6Ij7tM-a5iVPGboS5tmvVA")
+                .build();
+    }
+
+    @Test
+    public void shouldCreateSdJwtSuccessfully() {
+        SdJwt createdSdJwt = sdJwtFacade.createSdJwt(claimSet, disclosureSpec);
+
+        assertNotNull(createdSdJwt);
+    }
+
+    @Test
+    public void shouldVerifySdJwtSuccessfullyWithValidKeys() {
+        claimSet = TestUtils.readClaimSet(getClass(), "sdjwt/a1.example2-issuer-claims.json");
+
+        SdJwt sdJwt = sdJwtFacade.createSdJwt(claimSet, disclosureSpec);
+
+        List<SignatureVerifierContext> verifyingKeys = Collections.singletonList(
+                createSignatureVerifierContext("doc-signer-05-25-2022", "ES256", true)
+        );
+        IssuerSignedJwtVerificationOpts verificationOpts = createVerificationOptions();
+
+        try {
+            sdJwtFacade.verifySdJwt(sdJwt, verifyingKeys, verificationOpts);
+        } catch (VerificationException e) {
+            fail("Verification failed: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldReturnSdJwtString() {
+        SdJwt sdJwt = sdJwtFacade.createSdJwt(claimSet, disclosureSpec);
+
+        String sdJwtString = sdJwtFacade.getSdJwtString(sdJwt);
+
+        assertNotNull(sdJwtString);
```

Comment:
It migt be better to compare `sdJwt.toString()` and `sJwtString` and confirm that the both are exactly matched.